### ISSUE TITLE
Test Explorer not discovering all my xunit tests

### ISSUE THREAD

**Issue by antonblu** [formulahendry/vscode-dotnet-test-explorer#143]
Hello,

I have 0.5.3 test explorer running in VS Code 1.27.1.

My unit test solution is on an xunit framework with 15 tests on around 900 lines of code. Each 
test is a method in the following format:
[Theory]
[Inline Data(" ")]
public void test name number 1(variable type variable name)
{
test code;
}
[Theory]
[Inline Data(" ")]
public void test name number 2(variable type variable name)
{
test code;
}
[Theory]
[Inline Data(" ")]
public void test name number 3(variable type variable name)
{
test code;
}

etc etc etc

I saved my xunit tests then compiled them with no errors. 
Next , I clicked the 'refresh' icon in test explorer to discover my tests

I noticed that not all my tests were discovered, that is, not all of them appear in the test explorer panel in the left side.  Out of my 15 tests, only 10 are discovered in the test explorer left panel , while the remaining 5 are not discovered.

so I made sure these undiscovered tests run fine by going to the open cs file in the right side and clicking the 'Run Test' link right under the [Inline Data(" ")] line of each undiscovered test . Each run of the undiscovered test passed with no errors.

How can I get these undiscovered tests to appear in the test explorer panel in the left side ? 
Thank you for your time.
Anton

**Comment by stefanforsberg** [comment]
We have a known issue where tests using DisplayName are not shown, do you have any of those?

If it's not that it would be helpful if you could give us an actual test example that does not show up during discovery. If the current data is sensitive maybe you could obfuscate it somehow while still causing the error? What I'm looking for is something like this

``` csharp
[Theory]
[Inline Data("ThisIsMyDataThatCausesAnError")]
public void TestMethod(string data)
{
...
}
```

Just out of curiosity, if you press the "Run all tests" button, do the test previously not discovered show up in the tree?

![image](https://user-images.githubusercontent.com/358570/45403588-90a7df00-b65b-11e8-9e24-dd612d837fb2.png)

**Comment by antonblu** [comment]
Hi Stefan,

I ran the "Run all tests" button and I see the same tests in the tree. the undiscovered tests are still too shy to show up.
6 of the 10 discovered tests passed. i think the 4 that failed have stale inline data so i just need to look at the fails since i havent looked at it in 2 months.
the 5 undiscovered tests are still not showing in the tree in the test explorer panel in the left side.

with regard to displayname, if I understand you right. there are no displaynames after[Theory].
My tests are in the following exact format:

[Theory]
[InlineData("data1", "data2","data3")]
public void testmethodname(string variable1, string variable2, string variable3)
{
sel.navigate("homepage.aspx");
sel.verifyvisible("elementID");
sel.click("lblbutton1");
}

NOTE: we use sel as a shortcut custom command to refer to a helper file in the same solution that has the more complicated c# routines or lines to make it easier for the qa testers to write the tests-the gory code is hidden :) )
Thanks for asking....
Anton

**Comment by stefanforsberg** [comment]
Hmm ok, I suspected that it was perhaps related to having multiple inline data paramaters but this method works for me:

``` csharp
[Theory]
[InlineData("data1", "data2","data3")]
public void MultipleInlineData(string var1, string var2, string var3)
{
      $"{var1}{var2}{var3}".ShouldBe("data1data2data3");
}
```
![image](https://user-images.githubusercontent.com/358570/45443865-3d1eab00-b6c6-11e8-9474-95262f827b40.png)

If you could produce a stand alone project that replicates your error it would be helpful.

**Comment by antonblu** [comment]
Hi Stefan !
Using the 'inline data' lead you provided, I have figured out the culprit. if there is a  period as in '.' in the inline data such as 5.50 or  "Order saved successfully." the test explorer will not discover the test. For 5.5, the workaround that worked is 11/2  , and for the sentence with the dot, I will just rem it out for now. Thanks Stefan.

**Comment by stefanforsberg** [comment]
Hey @antonblu!

Ah okey, that sounds reasonable. We actually have another issue that I think might be the same as this #104. No solution for it yet though I'm afraid, but now we have a way of reproducing it at least so thanks for that!

**Comment by stefanforsberg** [comment]
After a closer look this problem stems from the fact that we use the . as a namespace seperator. So if you have MyNamespace.MyClass.MyMethod(value: "something.hello") it got confused by the dot inside the value since we just split the test name on the dot to build the tree. But we should be able to find some regexp magic to split on dots that are outside of parenthesis.

**Comment by stefanforsberg** [comment]
Should be fixed with 0.5.4
